students/Pirouz_N/lesson04/mailroom.py

```diff
@@ -0,0 +1,185 @@
+                       'Please make your selection:\n')
+        try:
+            option_dict[option]()
+        except KeyError:
+            print('Please only type 1 or 2 or 3 or 4 and hit enter. Any other option will not be acceptable.')
+
+
+def create_report():
+    """"This function will create a report."""
+    # Printing table
+    print('{:<30.30s}\t|{:^16.16s}\t|{:^12.12s}\t|{:^16.16s}'
+          .format('Donor Name', 'Total Given', 'Num Gifts', 'Average Gift'))
+    print('------------------------------------------------------------------------------------------------')
+
+    # printing donor table
+    for donor in donor_db:
+        # printing donor table
+        print(
+            '{:<30.30s}\t ${:>15.15s}\t{:>12.12s}\t ${:>15.15s}'
+            .format(donor, format(float(sum(donor_db[donor])), '.2f'), str(len(donor_db[donor])),
+                    format(float(sum(donor_db[donor]) / len(donor_db[donor])), '.2f')))
+    print('\n')
+
+
+def print_list_of_donors():
+    """"This function will print list of donors."""
+    print('{:<30.30s}\t'.format('Donor Name'))
+    print('---------------------------------')
+
+    # printing donor names
+    for donor in donor_db:
+        # printing donor name
+        print('{:<30.30s}\t'.format(donor))
+    print('\n')
+
+
+def write_thank_you(donername):
+    """"This function will write a thank to a donor.
+
+    Args:
+        donername: Is the name of the donor thank you card will be written to.
+
+    """
+    letter = 'Dear {},\n\n'.format(donername)
+    letter += 'Thank you for your latest donation of ${:0.2f}. With this donation your overall donation has reached'\
+        .format(donor_db[donername][len(donor_db[donername]) - 1])
+    letter += ' ${:0.2f}. We are very grateful for all your {} donations, and we appreciate all your support. '\
+        .format(float(sum(donor_db[donername])), len(donor_db[donername]))
+    letter += 'Please do not forget us in future we need because there is still more work that needs to be done and'
+    letter += ' we need your help to accomplish them.\n\n'
+    letter += 'Best Regards\n'
+    letter += 'Pirouz Naghavi'
+    return letter
+
+
+def send_thank_you_after_donation():
+    """"This function will type a thank card to the selected donor after a donation."""
+    while True:
+        option = input('Please type in the name of the donor you wish to thank.\n'
+                       'If you wish to see the list of all the donors, please type list.\n'
+                       'If you wish to add a new donor to the list, please type in their name.\n'
+                       'If you wish to return to the main menu please type exit and hit enter.\n'
+                       'Please type in your desired option and hit enter:\n')
+
+        if option == 'list':
+            print_list_of_donors()
+
+        elif option == 'exit':
+            break
+
+        else:
+            donor_names_list = [item for item in donor_db]
```

Comment:
a list from a dictionary can be created from `list(donor_db)` 

however, you don't need a list here. the correct dictionary usage here is to access via get() or via `[<key>]` syntax.

please correct.

Reply:
I am sorry. I don't know what I was thinking when I was writing this code. Corrected now!

Best Regards
Pirouz Naghavi